### DISCUSSION TITLE
Pre-loading stimuli

### DISCUSSION THREAD

**Discussion by keijiota** [jspsych/jsPsych#3139] [Q&A] [answered]
Hello jsPsych community, 
I think a normal protocol to load stimuli is to give a path to stimuli which are stored in a local machine. I just wonder if there is any approach to load stimuli which are stored in the website (as if we download packages on Github from command window) or some other local machine using IP address. The problem is that we want to keep our stimuli of image files in some place other than a local PC (which is actually data security system in our institution) and get access to there by some way. 
Thanks as always. 

Keiji

**Comment by jodeleeuw** [comment, marked as answer]
Hi @keijiota,

You can use a URL as a path for any image. This should work just fine:

```js
const preload = {
	type: jsPsychPreload,
	images: ['https://mystimuli.org/img1.png']
}
```

The examples in our docs use relative paths, so if the website is hosted online it'll just grab the files from the same machine that is acting as the webserver.

**Reply by hadarMenahem** [reply]
hi, this is working for regular websites (such as 'https://brainsandmachines.org/images/paper_figures/Distinguishing_representational_geometries_with_controversial.jpg'), but when I want to use an image from github it doesn't work anymore (I tried: 'https://github.com/hadarMenahem/project/blob/main/i05june05_static_street_boston_p1010764.jpeg'). 
do you have any idea why?
thanks

**Reply by Shaobin-Jiang** [reply]
That would be because `https://github.com/hadarMenahem/project/blob/main/i05june05_static_street_boston_p1010764.jpeg` does not point to an image. If you open that url, you would see that it is a page that displays the image as well as some other contents.

jsPsych does not know about this, however, as it simply assigns the url to the `src` attribute of an `Image`. The correct url would be `https://raw.githubusercontent.com/hadarMenahem/project/main/i05june05_static_street_boston_p1010764.jpeg`. You can find out about the correct url by right clicking on the image and copying its url.
